app/assets/javascripts/application.js

```diff
@@ -32,4 +33,50 @@ $(document).on("turbolinks:load", function() {
     }
   })
 
+  // Used to show the selected product and cost infomation in 'Select Product' step
+  $('#product-select').change(function(event){
+    var productId = $("#product-select").val();
+
+    $.ajax({
+      headers: {
+          'Accept' : 'application/vnd.api+json',
+          'Content-Type' : 'application/vnd.api+json'
+      },
+      url: Routes.product_path(productId),
+      type: 'GET',
+      success: function(data){
+        renderProductInformation(data);
+        renderCostInformation(data);
+      }
+    });
+  });
 });
+
+function renderProductInformation(data) {
```

Comment:
You'd probably be better off using a Javascript templating library for this sort of thing... check out Mustache, JST, EJS, many others. Can talk about tomorrow if you like.